libexec/rtld-elf/rtld_c18n.h

```diff
@@ -127,7 +143,11 @@ untrusted_stk_get(void)
 {
 	void *sp;
 
+#ifdef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
+	asm ("mov	%0, csp" : "=C" (sp));
```

Comment:
It looks like the only use of this doesn't care about the current address, but you need to be really careful when reading CSP because the compiler makes no guarantees about what value it will have (and if the asm isn't volatile then the assembly could be duplicated and see different addresses). I believe the current use doesn't matter, but I get nervous when C accesses CSP directly, rather than just having assembly pass it through to the C function like it used to.

Reply:
How about `cheri_getstack()`?

Reply:
Same issue, it's not about the assembly, it's about CSP's address being ill-defined within a function. The code was fine and worked, I can just see issues in future if you start to use untrusted_stk_get in other ways, it was just an observation about that. Certainly doing `mov %0, csp` followed by `sp = cheri_getstack()` as you now do doesn't make sense, that'll just always overwrite the output of the assembly and make it dead...

Reply:
Oops I somehow forgot to delete the asm line. But I agree that this could be dangerous in the future so I'll revert to the old way to passing csp through assembly.